src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -42,17 +42,11 @@ internal interface IPartition<TElement> : IIListProvider<TElement>
 
         IPartition<TElement> Take(int count);
 
-        TElement ElementAt(int index);
+        bool TryGetElementAt(int index, out TElement element);
```

Comment:
I measured on `Range()`, `OrderBy()`, `Empty()`, `Repeat()` and `OrderBy().Skip()`. `OrderBy()` came out about pretty much the same (not surprising since it was doing something like this internally anyway) `OrderBy().Skip()` a slight win, the rest a slight lose. All of them were very slight though, fraction-of-a-percent.